src/seedu/addressbook/data/person/Person.java

```diff
@@ -81,7 +82,25 @@ public int hashCode() {
 
     @Override
     public String toString() {
-        return getAsTextShowAll();
+        return getPrintableString(name, phone, email, address);
+        //return getAsTextShowAll();
+    }
+
+    public String getPrintableString(Printable... printables) {
+        final StringBuilder builder = new StringBuilder();
+        final String detailIsPrivate = "(private) ";
+
+        for(Printable p : printables){
+            builder.append(p.getPrintableString());
```

Comment:
Does it make sense to you that this method is placed into `Person` class this way? 
You're using both parameters and private values in this method which is "not a very well implementation".

Think about how to implement it better e.g. should this `getPrintableString(Printable... printables)` be in this class?
What if I've classes that implement `Printable` in future??

Can clarify with me in next tutorial if you have any doubts 😃